File: adaptive-arp-api-java/ARP/src/me/adaptive/arp/api/Cookie.java
/*
 *
 * =| ADAPTIVE RUNTIME PLATFORM |=======================================================================================
 *
 * (C) Copyright 2013-2014 Carlos Lozano Diez t/a Adaptive.me <http://adaptive.me>.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 * Original author:
 *
 *     * Carlos Lozano Diez
 *                 <http://github.com/carloslozano>
 *                 <http://twitter.com/adaptivecoder>
 *                 <mailto:carlos@adaptive.me>
 *
 * Contributors:
 *
 *     * Ferran Vila Conesa
 *                 <http://github.com/fnva>
 *                 <http://twitter.com/ferran_vila>
 *                 <mailto:ferran.vila.conesa@gmail.com>
 *
 *     * Aryslan
 *                 <http://github.com/Aryslan>
 *                 <http://twitter.com/Aryslan>
 *                 <mailto:ddbc@gft.com>
 *
 *     * Francisco Javier Martin Bueno
 *                 <https://github.com/kechis>
 *                 <mailto:kechis@gmail.com>
 *
 * =====================================================================================================================
 *
 */

package me.adaptive.arp.api;

/**
 * Created by FRMI on 12/08/2014.
 */
public class Cookie {
    /**
     * Name ot the cookie
     *
     * @since ARP1.0
     */
    private String name;
    /**
     * Value of the Cookie
     *
     * @since ARP1.0
     */
    private String value;

    /**
     * Constructor used by the implementation
     *
     * @param name
     * @param value
     * @since ARP1.0
     */
    public Cookie(String name, String value) {
        this.name = name;
        this.value = value;
    }

    /**
     * Returns the cookie name
     *
     * @return name
     * @since ARP1.0
     */
    public String getName() {
        return name;
    }

    /**
     * Set the cookie name
     *
     * @param name
     * @since ARP1.0
     */
    public void setName(String name) {
        this.name = name;
    }

    /**
     * Returns the cookie value
     *
     * @return
     * @since ARP1.0
     */
    public String getValue() {
        return value;
    }

    /**
     * Set the cookie value
     *
     * @param value
     * @since ARP1.0
     */
    public void setValue(String value) {
        this.value = value;
    }
}

File: adaptive-arp-api-objc/adaptive-arp-api-objc/ARP/Cookie.m
/*
 * =| ADAPTIVE RUNTIME PLATFORM |=======================================================================================
 *
 * (C) Copyright 2013-2014 Carlos Lozano Diez t/a Adaptive.me <http://adaptive.me>.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 * Original author:
 *
 *     * Carlos Lozano Diez
 *                 <http://github.com/carloslozano>
 *                 <http://twitter.com/adaptivecoder>
 *                 <mailto:carlos@adaptive.me>
 *
 * Contributors:
 *
 *     *
 *
 * =====================================================================================================================
 */

//
//  source: ../../../adaptive-arp-api-java/arp-api-specs/src/main/java/me/adaptive/arp/api/Cookie.java
//
//

#include "Cookie.h"

@implementation ARPCookie

- (id)initWithNSString:(NSString *)name
          withNSString:(NSString *)data {
  if (self = [super init]) {
    self->name_ = name;
    self->data_ = data;
  }
  return self;
}

- (id)init {
  return [super init];
}

- (NSString *)getDomain {
  return domain_;
}

- (void)setDomainWithNSString:(NSString *)domain {
  self->domain_ = domain;
}

- (NSString *)getPath {
  return path_;
}

- (void)setPathWithNSString:(NSString *)path {
  self->path_ = path;
}

- (NSString *)getScheme {
  return scheme_;
}

- (void)setSchemeWithNSString:(NSString *)scheme {
  self->scheme_ = scheme;
}

- (BOOL)isSecure {
  return secure_;
}

- (void)setSecureWithBoolean:(BOOL)secure {
  self->secure_ = secure;
}

- (long long int)getExpiry {
  return expiry_;
}

- (void)setExpiryWithLong:(long long int)expiry {
  self->expiry_ = expiry;
}

- (NSString *)getName {
  return name_;
}

- (void)setNameWithNSString:(NSString *)name {
  self->name_ = name;
}

- (NSString *)getData {
  return data_;
}

- (void)setDataWithNSString:(NSString *)data {
  self->data_ = data;
}

- (long long int)getCreation {
  return creation_;
}

- (void)copyAllFieldsTo:(ARPCookie *)other {
  [super copyAllFieldsTo:other];
  other->creation_ = creation_;
  other->data_ = data_;
  other->domain_ = domain_;
  other->expiry_ = expiry_;
  other->name_ = name_;
  other->path_ = path_;
  other->scheme_ = scheme_;
  other->secure_ = secure_;
}

+ (J2ObjcClassInfo *)__metadata {
  static J2ObjcMethodInfo methods[] = {
    { "initWithNSString:withNSString:", "Cookie", NULL, 0x1, NULL },
    { "init", "Cookie", NULL, 0x1, NULL },
    { "getDomain", NULL, "Ljava.lang.String;", 0x1, NULL },
    { "setDomainWithNSString:", "setDomain", "V", 0x1, NULL },
    { "getPath", NULL, "Ljava.lang.String;", 0x1, NULL },
    { "setPathWithNSString:", "setPath", "V", 0x1, NULL },
    { "getScheme", NULL, "Ljava.lang.String;", 0x1, NULL },
    { "setSchemeWithNSString:", "setScheme", "V", 0x1, NULL },
    { "isSecure", NULL, "Z", 0x1, NULL },
    { "setSecureWithBoolean:", "setSecure", "V", 0x1, NULL },
    { "getExpiry", NULL, "J", 0x1, NULL },
    { "setExpiryWithLong:", "setExpiry", "V", 0x1, NULL },
    { "getName", NULL, "Ljava.lang.String;", 0x1, NULL },
    { "setNameWithNSString:", "setName", "V", 0x1, NULL },
    { "getData", NULL, "Ljava.lang.String;", 0x1, NULL },
    { "setDataWithNSString:", "setData", "V", 0x1, NULL },
    { "getCreation", NULL, "J", 0x1, NULL },
  };
  static J2ObjcFieldInfo fields[] = {
    { "name_", NULL, 0x2, "Ljava.lang.String;", NULL,  },
    { "data_", NULL, 0x2, "Ljava.lang.String;", NULL,  },
    { "domain_", NULL, 0x2, "Ljava.lang.String;", NULL,  },
    { "path_", NULL, 0x2, "Ljava.lang.String;", NULL,  },
    { "scheme_", NULL, 0x2, "Ljava.lang.String;", NULL,  },
    { "secure_", NULL, 0x2, "Z", NULL,  },
    { "expiry_", NULL, 0x2, "J", NULL,  },
    { "creation_", NULL, 0x2, "J", NULL,  },
  };
  static J2ObjcClassInfo _ARPCookie = { "Cookie", "me.adaptive.arp.api", NULL, 0x1, 17, methods, 8, fields, 0, NULL};
  return &_ARPCookie;
}

@end
